src/MoBi.Core/OSPSuite.Core/Domain/Builder/CoreExpressionProfile.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections;
+using OSPSuite.Utility.Collections;
+using System.Collections.Generic;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Utility.Extensions;
+using static OSPSuite.Core.CoreConstants.ContainerName;
+
+namespace OSPSuite.Core.Domain.Builder
+{
+   public class CoreExpressionProfile : BuildingBlock, IEnumerable<ExpressionParameter>
+   {
+      private readonly ICache<IObjectPath, ExpressionParameter> _allValues;
+      public CoreExpressionProfile()
+      {
+         _allValues = new Cache<IObjectPath, ExpressionParameter>(x => x.Path, x => null);
+      }
+
+      public virtual string MoleculeName { set; get; }
```

Comment:
I don't think this is ok. We are inferring from the name so let's always retrieve from the name as well so that it's always consistent